docs/cookbook/prompt_engineering/text_based/self_ask.md

```diff
@@ -0,0 +1,248 @@
+            Intermediate answer: The area code 575 is located in New Mexico.
+            Follow up: When does monsoon season end in New Mexico?
+            Intermediate answer: Monsoon season in New Mexico typically ends in mid-September.
+            So the final answer is: mid-September.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What is the current official currency in the country where Ineabelle Diaz is a citizen?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which country is Ineabelle Diaz a citizen of?
+            Intermediate answer: Ineabelle Diaz is from Peurto Rico, which is in the United States of America.
+            Follow up: What is the current official currency in the United States of America?
+            Intermediate answer: The current official currency in the United States is the United States dollar.
+            So the final answer is: United States dollar.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="Where was the person who founded the American Institute of Public Opinion in 1935 born?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who founded the American Institute of Public Opinion in 1935?
+            Intermediate answer: George Gallup.
+            Follow up: Where was George Gallup born?
+            Intermediate answer: George Gallup was born in Jefferson, Iowa.
+            So the final answer is: Jefferson.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What language is used by the director of Tiffany Memorandum?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who directed the movie called Tiffany Memorandum?
+            Intermediate answer: Sergio Grieco.
+            Follow up: What language is used by Sergio Grieco?
+            Intermediate answer: Sergio Grieco speaks Italian.
+            So the final answer is: Italian.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What is the sports team the person played for who scored the first touchdown in Superbowl 1?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which player scored the first touchdown in Superbowl 1?
+            Intermediate answer: Max McGee.
+            Follow up: Which sports team did Max McGee play for?
+            Intermediate answer: Max McGee played for the Green Bay Packers.
+            So the final answer is: Green Bay Packers.
+            """
+        ),
+    ),
+]
+```
+
+Finally, we can use our Self-Ask implementation:
+
+```python
+response = self_ask_query(
+    query="The birth country of Jayantha Ketagoda left the British Empire when?",
```

Comment:
should we call the prompt engineered call by directly defining the query arg? Especially if its 1) super long and 2) we reuse it in a non engineered call. In this case, I've been going by naming the variable `prompt`, please confirm we're defining the argument directly and if not, if the variable name `prompt` is fine or you want something different

Reply:
I don't like the variable name "prompt" compared to it being the user's query which gets injected into the final prompt (that we're engineering).

It's fine if you want to define it as a variable so you can re-use it, I just wasn't doing that here and do defined it directly.

Reply:
Acutally revisiting this, I updated to handle the no example comparison and ended up refactoring. Now I agree I think having it as a var should be the default regardless going forward it looks nicer.